pkg/controlplane/controller/leaderelection/leaderelection_controller.go

```diff
@@ -278,7 +278,8 @@ func (c *Controller) reconcileElectionStep(ctx context.Context, leaseNN types.Na
 
 		if candidate.Spec.PingTime == nil ||
 			// If PingTime is outdated, send another PingTime only if it already acked the first one.
-			(candidate.Spec.PingTime.Add(electionDuration).Before(now) && candidate.Spec.PingTime.Before(candidate.Spec.RenewTime)) {
+			// This checks for pingTime <= renewTime because equality is possible in unit tests using a fake clock.
+			(candidate.Spec.PingTime.Add(electionDuration).Before(now) && !candidate.Spec.RenewTime.Before(candidate.Spec.PingTime)) {
```

Comment:
you had added a `+-time.MilliSecond` somewhere in the tests. Can that be removed with this change?

Reply:
removed

Reply:
ack, not affecting this PR, but checking for <= instead of < was specifically for this test since the clock was fixed. Now that we're using a real clock it's technically unnecessary?

Reply:
I think it's a little bit more correct as it is now. Though of course the clock is very fine grained. So maybe in practice it does not matter.